packages/sdk-react/src/components/userRoles/UserRoles.tsx

```diff
@@ -1,19 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 
-import { PageHeader } from '@/components';
+import { Dialog, PageHeader, UserRoleDetails } from '@/components';
 import { MoniteStyleProvider } from '@/core/context/MoniteProvider';
+import { useEntityUserByAuthToken } from '@/core/queries';
+import { useIsActionAllowed } from '@/core/queries/usePermissions';
+import { AccessRestriction } from '@/ui/accessRestriction';
 import { t } from '@lingui/macro';
 import { useLingui } from '@lingui/react';
+import { ActionEnum } from '@monite/sdk-api';
 
 import { UserRolesTable } from './UserRolesTable';
 
 export const UserRoles = () => {
   const { i18n } = useLingui();
+  const [isDetailsDialogOpened, setIsDetailsDialogOpened] = useState(false);
+  const [selectedUserRoleId, setSelectedUserRoleID] = useState<
+    string | undefined
+  >(undefined);
+  const { data: user } = useEntityUserByAuthToken();
+  const { data: isReadAllowed, isInitialLoading: isReadAllowedLoading } =
+    useIsActionAllowed({
+      method: 'role',
+      action: ActionEnum.READ,
+      entityUserId: user?.id,
+    });
+
+  const onRowClick = (id: string) => {
+    setIsDetailsDialogOpened(true);
+    setSelectedUserRoleID(id);
+  };
 
   return (
     <MoniteStyleProvider>
       <PageHeader title={t(i18n)`User Roles`} />
-      <UserRolesTable />
+
+      {!isReadAllowed && !isReadAllowedLoading && <AccessRestriction />}
```

Comment:
I think you missed the error state in this case.

Reply:
currently we don't render error state in supercomponents

Reply:
But what happens if the server responds with an error?

Reply:
The error toggle will be shown and the table will be empty.
I agree that it is not the ideal behavior, but it is a common approach in our super components.

Reply:
Could you please create a ticket that describes this behavior?
We should improve it because an understandable error is better than an empty table.

Reply:
Got it, thanks